keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -104,7 +108,54 @@ class ApplicationService(
     }
   }
 
-  private fun getFirstDeliveryConfigFor(application: String): DeliveryConfig? =
+  /**
+   * Takes an artifact version, plus information about the type of artifact, and constructs a summary view.
+   * This should be supplemented/re-written to use actual data from stash/git/etc instead of parsing everything
+   * from the version string.
+   */
+  private fun versionToSummary(
+    artifact: DeliveryArtifact,
+    version: String,
+    environments: Set<ArtifactSummaryInEnvironment>
+  ): ArtifactVersionSummary =
+    when (artifact.type) {
+      deb -> {
+        val appversion = AppVersion.parseName(version)
+        ArtifactVersionSummary(
+          version = version,
+          environments = environments,
+          displayName = appversion?.version ?: version.removePrefix("${artifact.name}-"),
```

Comment:
I was hoping that we might be able to extract the "clean" version string just using the regex in the `TagVersionStrategy`. Is there a reason that wouldn't work?

Reply:
This here is for a debian, and using frigga to parse the app version is definitely the right call here.

Re docker images: my first pass of this means that the 'friendly name' is the full tag. I thought about using the sorting regex to pull out info, but I think that is more hairy. I expect most of this code here to be replaced with a better way of populating the build id and commit, so I want to keep it as simple as possible

Reply:
Thanks for explaining!